test/multiplexed-session.ts

```diff
@@ -0,0 +1,326 @@
+      _createSessionStub = sandbox
+        .stub(multiplexedSession, '_createSession')
+        .resolves();
+    });
+
+    it('should create mux session', () => {
+      multiplexedSession.createSession();
+      assert.strictEqual(_createSessionStub.callCount, 1);
+    });
+
+    it('should start housekeeping', done => {
+      multiplexedSession.createSession();
+      setImmediate(() => {
+        try {
+          assert.strictEqual(_maintainStub.callCount, 1);
+          done();
+        } catch (err) {
+          done(err);
+        }
+      });
+    });
+
+    it('should not propagate instance and database not found errors for Multiplexed Session', () => {
+      const multiplexedSession = new MultiplexedSession(DATABASE);
+      const error = {
+        code: Status.NOT_FOUND,
+        message: 'Database not found',
+      } as grpc.ServiceError;
+      sandbox.stub(multiplexedSession, '_createSession').rejects(error);
+      try {
+        multiplexedSession.createSession();
+      } catch (e) {
+        shouldNotBeCalled();
+      }
+    });
+
+    it('should not propagate errors for Multiplexed Session when default credentials not set', () => {
+      const multiplexedSession = new MultiplexedSession(DATABASE);
+      const error = {
+        message: 'Could not load the default credentials',
+      } as grpc.ServiceError;
+      sandbox.stub(multiplexedSession, '_createSession').rejects(error);
+      try {
+        multiplexedSession.createSession();
+      } catch (e) {
+        shouldNotBeCalled();
+      }
+    });
+
+    it('should not propagate errors for Multiplexed Session when projectId not set', () => {
+      const multiplexedSession = new MultiplexedSession(DATABASE);
+      const error = {
+        code: Status.NOT_FOUND,
+        message: 'Unable to detect a Project Id in the current environment',
+      } as grpc.ServiceError;
+      sandbox.stub(multiplexedSession, '_createSession').rejects(error);
+      try {
+        multiplexedSession.createSession();
+      } catch (e) {
+        shouldNotBeCalled();
+      }
+    });
+
+    it('should propagate errors for Multiplexed Session which gets emitted', async () => {
+      const multiplexedSession = new MultiplexedSession(DATABASE);
+      const fakeError = new Error();
+      sandbox.stub(multiplexedSession, '_createSession').rejects(fakeError);
+      const errorPromise = new Promise<void>((resolve, reject) => {
+        multiplexedSession.once('error', err => {
+          if (err === fakeError) {
+            resolve();
+          } else {
+            reject(new Error('Unexpected error emitted'));
+          }
+        });
+      });
+
+      multiplexedSession.createSession();
+
+      await errorPromise;
+    });
+  });
+
+  describe('_maintain', () => {
+    it('should set an interval to refresh mux sessions', done => {
+      const expectedInterval =
+        multiplexedSession.refreshRate! * 24 * 60 * 60000;
+      const clock = sandbox.useFakeTimers();
```

Comment:
make sure to clear the clock after the test is over. Probably you can do below 
let clock;
beforeEach(() => {
  clock = sandbox.useFakeTimers();
});
afterEach(() => {
  clock.restore();
});

Reply:
done